File: frontend/src/app/shared/UserDto.ts
import { GroupDto } from './GroupDto'
import { IDisplayName } from './IDisplayName'

export class UserDto implements IDisplayName {
  id: number
  email: string
  userName: string
  type: string
  groups: GroupDto[] = []
  features: number[] = []

  public displayName(): string {
    return this.userName || this.email
  }

}

File: backend/src2/SharedModels/DTOs/UserDto.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using SharedModels.Enum;

namespace SharedModels.DTOs
{
    public class UserDto
    {
        public int Id { get; set; }
        public string Email { get; set; }
        public string UserName { get; set; }

        [JsonConverter(typeof(StringEnumConverter))]
        public UserType Type { get; set; }

        public List<GroupDto> Groups { get; set; }
    }
}